ui.frontend/src/view/InstanceManager.js

```diff
@@ -154,10 +154,11 @@ export default class InstanceManager {
 
     #registerInstanceHandlers(childView) {
         //doesn't support repeatable panel inside repeatable panel
-        Utils.registerClickHandler(childView.element.parentElement, Constants.DATA_HOOK_ADD_INSTANCE, (event) => {
+        const parentElement = typeof this?.repeatableParentView?.getRepeatableRootElement === "function" ? this.repeatableParentView.getRepeatableRootElement(childView) : childView.element.parentElement;
+        Utils.registerClickHandler(parentElement, Constants.DATA_HOOK_ADD_INSTANCE, (event) => {
```

Comment:
this will break for repeatable accordion inside repeatable accordion, till now we were not consuming this private function, so we were good. But if we intend to use this then repeatability inside repeatability needs to be fixed and corresponding test case needs to be added.

Reply:
We are not changing the existing logic here. We are simply adding a new function (getRepeatableRootElement) that will return the root elements.